### PR TITLE
remove obsolete models, add missing info about RNN-T

### PR DESCRIPTION
**Pull request by mwawrzos**

### CODE REVIEW

File: training_rules.adoc [hunk 1]
```diff
@@ -78,21 +78,12 @@ The benchmark suite consists of the benchmarks shown in the following table.
 | |Image segmentation (medical) |KiTS19
 | |Object detection (light weight) |COCO
 | |Object detection (heavy weight) |COCO
-|Language |Translation (recurrent) |WMT English-German
-| |Translation (non-recurrent) |WMT English-German
+|Language |Speech recognition |LibriSpeech
```
Comment (qpjaada) [inline training_rules.adoc:81]: Should `Speech` be a separate area from `Language` ?

File: training_rules.adoc [hunk 2]
```diff
@@ -463,8 +436,7 @@ Each benchmark result is based on a set of run results. The number of results fo
 | |Image segmentation (medical) | 20
 | |Object detection (light weight) |5
 | |Object detection (heavy weight) |5
-|Language |Translation (recurrent) |10
-| |Translation (non-recurrent) |10
+|Language |Speech recognition |10
```
Comment (qpjaada) [inline training_rules.adoc]: Not sure what the `Speech recognition (recurrent)` entry in the row below refers to?

**Review by qpjaada — COMMENTED** [review]
Wondering whether we should have `Speech` as a area separate from `Language`.

Also, not sure what the entry `Speech recognition (recurrent)` in the table of number of runs for benchmark results refers to.

File: training_rules.adoc (hunk 1 above)
Comment (mwawrzos) [reply training_rules.adoc:81]: That was my first idea, but earlier it was under `Language`: https://github.com/mlcommons/training_policies/pull/438/files#diff-7f40dc8339752524c530b449806bdd880ed3dc21d2e43c220193b6660f316434L93

@johntran-nv, @petermattson, what do you think?

File: training_rules.adoc (hunk 2 above)
Comment (mwawrzos) [reply training_rules.adoc]: Ah, true. I missed that, thanks

Comment (mwawrzos) [reply training_rules.adoc]: fixed

File: training_rules.adoc (hunk 1 above)
Comment (johntran-nv) [reply training_rules.adoc:81]: I think either way would be fine, personally.  So I'm marking this as resolved so we can make forward progress.

**Comment by johntran-nv** [comment]
@qpjaada are you satisfied with this PR now that Marek has addressed your questions?

+emizan@google.com, could you also review/approve?

**Comment by qpjaada** [comment]
> @qpjaada are you satisfied with this PR now that Marek has addressed your questions?
> 
I am fine with these changes. If there are no objections - lets have a `Speech` area. Approving.

**Review by qpjaada — APPROVED** [review]
Approving - fine with either having a `Speech` area or not.

**Review by emizan76 — APPROVED** [review]

**Pull request merged**